apps/trackers/jira/query.py

```diff
@@ -112,19 +119,40 @@ def generate_description(self):
         """
         self._query["fields"]["description"] = self.description
 
-    # TODO we should not delete other labels
-    # because the engineeting may use them
     def generate_labels(self):
         """
         generate query for Jira labels
         """
-        self._query["fields"]["labels"] = [
-            "SecurityTracking",
-            "Security",
-            f"pscomponent:{self.ps_component}",
-        ] + list(  # add all linked non-empty CVE IDs
-            self.tracker.affects.exclude(flaw__cve_id__isnull=True).values_list(
-                "flaw__cve_id", flat=True
+
+        all_existing_labels = self.tracker.meta_attr.get("labels", [])
+
+        # These labels will be edited here by recreating them as needed.
+        labels_allowed_to_edit_here = set()
+        for lbl in all_existing_labels:
+            if (
+                lbl.startswith("pscomponent:")
+                or lbl in ["SecurityTracking", "Security", "validation-requested"]
+                or CVE_NAME_RE.match(lbl)
+            ):
+                labels_allowed_to_edit_here.add(lbl)
+
+        # These labels are from elsewhere than this method and preserved with their ordering intact.
+        # Because the engineering may use them.
+        labels_to_preserve = [
+            x for x in all_existing_labels if x not in labels_allowed_to_edit_here
+        ]
```

Comment:
`labels_allowed_to_edit_here` are not necessary for the rest of the code, so there is no need to have them.
We are interested only in `labels_to_preserve`, which can be created directly from `all_existing_labels` in one loop and condition (if I am right, then `all_existing_labels` = `labels_allowed_to_edit_here` + `labels_to_preserve`).

Reply:
You are correct in all points. I don't understand how that is better or more understandable, but did as you said. (I presume you didn't just want to state truthful statements, but wanted me to change it, right?) Please review commit [2c8f57f](https://github.com/RedHatProductSecurity/osidb/pull/428/commits/2c8f57f136f522f3cffc42df037bee5920c849ae)

Reply:
Initially, there was unclear sorting, two loops and two variables. Now, there is one loop, one variable and straightforward logic (with the code of half-length). I think the updated version is definitely better. Thank you!